### PR TITLE
[SYCL] Optimize includes in SYCL headers (part 1)

### PR DESCRIPTION
**Pull request by alexbatashev**
Eliminate includes of `iostream`, `sstream`, `fstream`, and `cmath` and remove some outdated declarations. Make sure aforementioned headers are never included again (results in ~10% parsing speed improvement).

### CODE REVIEW

**Comment by alexbatashev** [comment]
Split this PR into two pieces as suggested by @romanovvlad. Second (but not the last) patch is https://github.com/intel/llvm/pull/4642.

File: sycl/include/CL/sycl/detail/common.hpp [hunk 1]
```diff
@@ -168,7 +117,7 @@ template <class Obj> decltype(Obj::impl) getSyclObjImpl(const Obj &SyclObject) {
 // must make sure the returned pointer is not captured in a field or otherwise
 // stored - i.e. must live only as on-stack value.
 template <class T>
-typename detail::add_pointer_t<typename decltype(T::impl)::element_type>
+typename std::add_pointer_t<typename decltype(T::impl)::element_type>
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/common.hpp:120]: Isn't it a part of another PR?

Comment (alexbatashev) [reply sycl/include/CL/sycl/detail/common.hpp:120]: Right, a few things might have slipped in here. It was hard to split it cleanly.

**Review by romanovvlad — CHANGES_REQUESTED** [review]
Please, resolve pre-commit fails.

**Pull request closed without merging**